### PR TITLE
Migrate the github workflow dependencies-update to tekton

### PR DESCRIPTION
**Pull request by Roming22**
Signed-off-by: Romain Arnaud <rarnaud@redhat.com>

Tested successfully [when a PR is already opened](https://console-openshift-console.apps.pipeline-stage.3jhu.p1.openshiftapps.com/k8s/ns/pipeline-service-ci/tekton.dev~v1beta1~PipelineRun/pipeline-service-update-dependencies-cbdq2) and [when a PR needs to be opened](https://console-openshift-console.apps.pipeline-stage.3jhu.p1.openshiftapps.com/k8s/ns/pipeline-service-ci/tekton.dev~v1beta1~PipelineRun/pipeline-service-update-dependencies-jt7pq).

### CODE REVIEW

File: .tekton/update-dependencies.yaml [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                    secretKeyRef:
+                      name: github
+                      key: token
+              script: |
+                #!/usr/bin/env bash
+                set -o errexit
+                set -o pipefail
+                set -x
+                echo OK
+                cd "$(workspaces.source.path)"
```
Comment (xinredhat) [inline .tekton/update-dependencies.yaml:124]: You could specify the property `workingDir="(workspaces.source.path)"` at this step, then you can remove this line.  for example
```
- name: is-pr-required
   image: quay.io/redhat-pipeline-service/dependencies-update:main
   imagePullPolicy: Always
   workingDir="(workspaces.source.path)"
```

File: .tekton/update-dependencies.yaml [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+          steps:
+            - name: setup-local-repository
+              image: quay.io/redhat-pipeline-service/dependencies-update:main
+              imagePullPolicy: Always
+              script: |
+                #!/usr/bin/env bash
+                set -o errexit
+                set -o pipefail
+                set -x
+                cd "$(workspaces.source.path)"
```
Comment (xinredhat) [inline .tekton/update-dependencies.yaml:81]: You could specify the property workingDir="(workspaces.source.path)" at step, then you can remove this line.

File: .tekton/update-dependencies.yaml [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+                git checkout -b "$(params.source_branch)"
+            - name: update-dockerfiles
+              image: quay.io/redhat-pipeline-service/dependencies-update:main
+              imagePullPolicy: Always
+              script: |
+                #!/usr/bin/env bash
+                set -o errexit
+                set -o pipefail
+                set -x
+                cd "$(workspaces.source.path)"
```
Comment (xinredhat) [inline .tekton/update-dependencies.yaml:103]: same as above

File: .tekton/manifests/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+  - namespace.yaml
+  - repository.yaml
+  - https://raw.githubusercontent.com/tektoncd/catalog/main/task/git-clone/0.9/git-clone.yaml
+  - https://raw.githubusercontent.com/tektoncd/catalog/main/task/github-open-pr/0.2/github-open-pr.yaml
```
Comment (xinredhat) [inline .tekton/manifests/kustomization.yaml:9]: The both tasks `git-clone.yaml` and `github-open-pr` are build-in ClusterTask in Openshift-pipeline. Do we have to add them?

Comment (Roming22) [reply .tekton/manifests/kustomization.yaml:9]: It wouldn't work before I added them. Based on the conversations we had, I was expecting the ClusterTask would be retrieved automatically from the tekton catalog, but that is not the behavior I experienced.

File: .tekton/update-dependencies.yaml (hunk 1 above)
Comment (Roming22) [reply .tekton/update-dependencies.yaml]: Done

File: .tekton/update-dependencies.yaml (hunk 2 above)
Comment (Roming22) [reply .tekton/update-dependencies.yaml]: Done

File: .tekton/update-dependencies.yaml (hunk 3 above)
Comment (Roming22) [reply .tekton/update-dependencies.yaml]: Done

File: .tekton/manifests/kustomization.yaml (hunk 4 above)
Comment (xinredhat) [reply .tekton/manifests/kustomization.yaml:9]: I just realized that Pipelines as Code support fetching remote tasks or pipeline from a remote location with annotations on PipelineRun. For example
```
pipelinesascode.tekton.dev/task: "[git-clone, github-open-pr]"
```

Comment (Roming22) [reply .tekton/manifests/kustomization.yaml:9]: @xinredhat Fixed

**Pull request merged**